contracts/token/ERC1155/IERC1155.sol

```diff
@@ -5,27 +5,92 @@ pragma solidity ^0.6.2;
 import "../../introspection/IERC165.sol";
 
 /**
-    @title ERC-1155 Multi Token Standard basic interface
-    @dev See https://eips.ethereum.org/EIPS/eip-1155
+ * @dev Required interface of an ERC1155 compliant contract, as defined in the
+ * https://eips.ethereum.org/EIPS/eip-1155[EIP].
  */
 interface IERC1155 is IERC165 {
+    /**
+     * @dev Emitted when `value` tokens of token type `id` are transfered from `from` to `to` by `operator`.
+     */
     event TransferSingle(address indexed operator, address indexed from, address indexed to, uint256 id, uint256 value);
 
+    /**
+     * @dev Equivalent to multiple {TransferSingle} events, where `operator`, `from` and `to` are the same for all
+     * transfers.
+     */
     event TransferBatch(address indexed operator, address indexed from, address indexed to, uint256[] ids, uint256[] values);
 
+    /**
+     * @dev Emitted when `account` grants or revokes permission to `operator` to transfer their tokens, according to
+     * `approved`.
+     */
     event ApprovalForAll(address indexed account, address indexed operator, bool approved);
 
+    /**
+     * @dev Emitted when the URI for token type `id` changes to `value`, if it is a non-programmatic URI.
```

Comment:
We should explain the way the `URI` event and `uri()` function interact somewhere, and have a link to that here.